js/restaurant.js

```diff
@@ -431,28 +460,65 @@ function fireRule(rule) {
   }
 
   if(win){
+
+    if (localStorage.getItem("hardcore") === "1" && progress.guessHistory[currentLevel]) {
+      for (let s of document.querySelectorAll('.display-help .solutions *')) {
+        if (s.textContent === rule) {
+
+          $(".strobe").removeClass("strobe");
+          ruleSelected.removeClass("strobe");
+          ruleSelected.addClass("shake");
+
+          s.classList.add("shake")
+          setTimeout(function(){
+            s.classList.remove("shake");
+            $(".shake").removeClass("shake");
+            $(".strobe").removeClass("strobe");
+            levelSelected.addClass("strobe");
+          },500);
+
+          $(".result").fadeOut();
+
+          return;
+        }
+      }
+    }
```

Comment:
If the solution exists find it in the solutions container and shake it to indicate it's there